fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -38,12 +39,131 @@
 UNEXPECTED_DATA_TYPE_ERROR_MESSAGE = "Unexpected data type {} encountered."
 ESTIMATOR_ERROR_MESSAGE = "'estimator' cannot be `None`."
 
-_SUPPORTED_CONSTRAINTS = [DEMOGRAPHIC_PARITY, EQUALIZED_ODDS]
-
 logger = logging.getLogger(__name__)
 
+# All allowed utilities must have values between 0 and 1 and attain both extremes as
+# the threshold goes from -Inf to Inf.
+SUPPORTED_UTILITIES = {
+    'selection_rate',
+    'demographic_parity',
+    'false_positive_rate',
+    'false_negative_rate',
+    'true_positive_rate',
+    'true_negative_rate',
+}
+
+# For equalized odds, we only allow objectives that are non-decreasing in true_positives,
+# holding n, positives, negatives, true_negatives, and false_positives fixed.
+SUPPORTED_OBJECTIVES_FOR_EQUALIZED_ODDS = {
+    'accuracy_score',
+}
+
+SUPPORTED_CONSTRAINTS = [*SUPPORTED_UTILITIES, EQUALIZED_ODDS]
+NOT_SUPPORTED_CONSTRAINTS_ERROR_MESSAGE = (
+    "Currently only the following constraints are supported: {}.".format(
+        ", ".join(SUPPORTED_CONSTRAINTS)))
+NOT_SUPPORTED_OBJECTIVES_ERROR_MESSAGE = (
+    "Currently only the following objectives are supported: {}.".format(
+    ", ".join(METRIC_DICT.keys())))
+NOT_SUPPORTED_OBJECTIVES_FOR_EQUALIZED_ODDS_ERROR_MESSAGE = (
+    "Currently only the following objectives are supported: {}.".format(
+    ", ".join(SUPPORTED_OBJECTIVES_FOR_EQUALIZED_ODDS)))
+
+class InterpolatedThresholder(BaseEstimator, MetaEstimatorMixin):
```

Comment:
I'm not sure I like that it's in the same file. Makes the entire file more crowded with things. :-(
So far, if I search for `fit` in this file I know where I end up. Now I have to scroll back up and search for the class name.

Reply:
I don't mind pulling out into a separate file.